src/Corax/Querying/Matches/MemoizationMatch.Provider.cs

```diff
@@ -125,13 +112,29 @@ private void InitializeInner()
             End:
             // The problem is that multiple Fill calls do not ensure that we will get a sequence of ordered
             // values, therefore we must ensure that we get a 'sorted' sequence ensuring those happen.
+            SetSortingMode();
```

Comment:
This is the only usage of this method. Wouldn't it be more readable if we had it like:
```
if (_sortMode == SortMode.Default)
{
    var skipSorting = _inner.AttemptToSkipSorting();

    _sortMode = skipSorting switch
    {
        SkipSortingResult.ResultsNativelySorted => SortMode.Skip, // if the inner already sorted, we don't need
        SkipSortingResult.WillSkipSorting => SortMode.Required, // if the inner skipped sorting, we have to
        SkipSortingResult.SortingIsRequired => SortMode.Required,
        _ => throw new ArgumentOutOfRangeException(skipSorting.ToString())
    };
}
else if (_sortMode == SortMode.Required)
{
    // We need to sort and remove duplicates.
    count = Sorting.SortAndRemoveDuplicates(Buffer[..count]);
}